proto/api/v1/auth_service.proto

```diff
@@ -20,21 +21,146 @@ message UserAttribute {
 
 message AuthStatus {
     oneof id {
-        string                user_id     = 1;
-        storage.ServiceIdentity       service_id  = 2;
+        string                  user_id         = 1;
+        storage.ServiceIdentity service_id      = 2;
     }
-    google.protobuf.Timestamp expires     = 3;
-    string                    refresh_url = 4;
+    google.protobuf.Timestamp   expires         = 3;
+    string                      refresh_url     = 4;
+    storage.AuthProvider        auth_provider   = 5;
+    storage.UserInfo            user_info       = 6;
+    repeated UserAttribute      user_attributes = 7;
+}
+
+// AuthMachineToMachineConfig determines rules for exchanging an identity token from a third party with
+// a Central access token. The M2M stands for machine to machine, as this is the intended use-case
+// for the config.
+message AuthMachineToMachineConfig {
+    // UUID of the config.
+    string id = 1;
+
+    // The type of the auth machine to machine config.
+    // Currently supports GitHub actions or any other generic OIDC provider to use for verifying and
+    // exchanging the token.
+    enum Type {
+        GITHUB_ACTIONS = 0;
+        GENERIC        = 1;
+    }
+    Type type          = 2;
+
+    // The expiration of the token annotated as a duration.
+    // Possible valid time units are: s, m, h.
+    // The maximum allowed expiration duration is 24h.
+    // As an example: 2h45m.
+    string token_expiration_duration = 3;
+
+   // Mappings map an identity token's claim values to a specific role within Central.
+    message Mapping {
+        // A key within the identity token's claim value to use.
+        string key            = 1;
+        // The value that should match to map the identity token to a specific role.
+        // This supports regular expressions of RE2 format, see more here:
+        // https://github.com/google/re2/wiki/Syntax.
+        string value          = 2;
+        // The role which should be issued when the key and value match for a particular identity token.
+        string role           = 3;
```

Comment:
Ah, good catch, I missed adding this.

What I did:
- Added within the validation to verify the referenced role actually exists. If it does not exist, it will be rejected.
- Added a FK to the `Role` field with the constraint of `restrict-delete`. In case a role is removed, but still referenced in a config, deletion of the role will fail. This mimics the current behavior w.r.t. groups referencing roles [which is a custom implementation at the moment](https://github.com/stackrox/stackrox/blob/8ea31d953e93952ac8b3c09d31d3ed7349f143ec/central/role/datastore/datastore_impl.go#L864-L877) (filed [ROX-20307](https://issues.redhat.com/browse/ROX-20307) to change this, will be done in a follow-up).

Reply:
When this case occurs, does the error message returned make it evident that the role could not be deleted because it is in use?